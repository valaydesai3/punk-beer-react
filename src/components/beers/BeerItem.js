import React from 'react';
import PropTypes from 'prop-types';
import NoImage from '../layout/image-not-available.png';

const BeerItem = ({ beer }) => {
  return (
    <div className="card beer-item" data-testid="beerItem">
      <div className="card-grid-img">
        <img src={beer.image_url === null ? NoImage : beer.image_url} alt="Beer" />
      </div>
      <div className="card-grid-body">
        <p className="card-grid-name">{beer.name}</p>
        <p className="card-grid-description">{beer.description}</p>
        <p className="card-grid-tagline">
          {beer.tagline}
          <span>{beer.first_brewed}</span>
          <span>{beer.abv}&#37;</span>
        </p>
      </div>
    </div>
  );
};

BeerItem.propTypes = {
  beer: PropTypes.object.isRequired
};

export default BeerItem;
