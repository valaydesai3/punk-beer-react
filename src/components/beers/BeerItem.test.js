// __tests__/fetch.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import BeerItem from './BeerItem';

afterEach(cleanup);

test('renders beer item', async () => {
  const randomBeer = {
    name: 'Chaos Theory',
    tagline: 'Predictably Random IPA.',
    first_brewed: '10/2008',
    description:
      'Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.',
    image_url: 'https://images.punkapi.com/v2/124.png',
    abv: 7.1
  };
  const { getByTestId } = render(<BeerItem beer={randomBeer} />);
  expect(getByTestId('beerItem')).toHaveTextContent('Chaos Theory');
  expect(getByTestId('beerItem')).toHaveTextContent(
    'grapefruit, pineapple and caramel sing above the chaos of life.'
  );
  expect(getByTestId('beerItem')).toHaveTextContent('Predictably Random IPA.');
  expect(getByTestId('beerItem')).toHaveTextContent('10/2008');
  expect(getByTestId('beerItem')).toHaveTextContent('7.1%');
  expect(getByTestId('beerItem')).toMatchSnapshot();
});
