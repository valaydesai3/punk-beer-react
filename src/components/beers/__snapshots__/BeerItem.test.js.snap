// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders beer item 1`] = `
<div
  class="card beer-item"
  data-testid="beerItem"
>
  <div
    class="card-grid-img"
  >
    <img
      alt="Beer"
      src="https://images.punkapi.com/v2/124.png"
    />
  </div>
  <div
    class="card-grid-body"
  >
    <p
      class="card-grid-name"
    >
      Chaos Theory
    </p>
    <p
      class="card-grid-description"
    >
      Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.
    </p>
    <p
      class="card-grid-tagline"
    >
      Predictably Random IPA.
      <span>
        10/2008
      </span>
      <span>
        7.1
        %
      </span>
    </p>
  </div>
</div>
`;
